src/Datasource/RepositoryInterface.php

```diff
@@ -148,7 +152,7 @@ public function exists($conditions): bool;
      * @param \ArrayAccess|array $options The options to use when saving.
      * @return \Cake\Datasource\EntityInterface|false
      */
-    public function save(EntityInterface $entity, $options = []);
+    public function save(EntityInterface $entity, ArrayAccess|array $options = []): EntityInterface|false;
```

Comment:
Shouldnt we go for nullable stuff in 5.0? For a cleaner API?

Reply:
Allowing null here would require adding an extra check for null value or cast to array. So that would be just overhead without any benefit.

Reply:
As return value?
I dont quite follow.

Reply:
Ah, for the return value, yes I guess that is an option.

Is returning the entity even necessary? Since it's an object it's the same entity reference that's passed as an argument.  I am not aware of a use case where the returned value is a reference to a new entity object. In which case the return value could be just boolean.

Reply:
Are entities one day immutable? Probably not